src/OrleansAzureUtils/MultiClusterNetwork/GossipTableInstanceManager.cs

```diff
@@ -0,0 +1,327 @@
+            //Debug.Assert(entry.RowKey == GossipTableEntry.CONFIGURATION_ROW);
+
+            entry.GossipTimestamp = configuration.AdminTimestamp;
+            entry.Clusters = string.Join(GossipTableEntry.ClustersListSeparator, configuration.Clusters);
+            entry.Comment = configuration.Comment ?? "";
+
+            return (await TryUpdateTableEntryAsync(entry, eTag).ConfigureAwait(false) != null);
+        }
+
+        internal async Task<bool> TryCreateGatewayEntryAsync(GatewayEntry entry)
+        {
+            var row = new GossipTableEntry()
+            {
+                PartitionKey = GlobalServiceId,
+                RowKey = GossipTableEntry.ConstructRowKey(entry.SiloAddress, entry.ClusterId),
+                Status = entry.Status.ToString(),
+                GossipTimestamp = entry.HeartbeatTimestamp
+            };
+
+            return (await TryCreateTableEntryAsync(row).ConfigureAwait(false) != null);
+        }
+
+
+        internal async Task<bool> TryUpdateGatewayEntryAsync(GatewayEntry entry, GossipTableEntry row, string eTag)
+        {
+            //Debug.Assert(row.ETag == eTag);
+            //Debug.Assert(row.PartitionKey == GlobalServiceId);
+            //Debug.Assert(row.RowKey == GossipTableEntry.ConstructRowKey(entry.SiloAddress, entry.ClusterId));
+            
+            row.Status = entry.Status.ToString();
+            row.GossipTimestamp = entry.HeartbeatTimestamp;
+
+            return (await TryUpdateTableEntryAsync(row, eTag).ConfigureAwait(false) != null);
+        }
+
+        internal Task<bool> TryDeleteGatewayEntryAsync(GossipTableEntry row, string eTag)
+        {
+            return TryDeleteTableEntryAsync(row, eTag);
+        }
+
+        internal async Task<int> DeleteTableEntries()
+        {
+            var entries = await storage.ReadAllTableEntriesForPartitionAsync(GlobalServiceId).ConfigureAwait(false);
+            var entriesList = new List<Tuple<GossipTableEntry, string>>(entries);
+            if (entriesList.Count <= AzureTableDefaultPolicies.MAX_BULK_UPDATE_ROWS)
+            {
+                await storage.DeleteTableEntriesAsync(entriesList).ConfigureAwait(false);
+            }
+            else
+            {
+                List<Task> tasks = new List<Task>();
+                foreach (var batch in entriesList.BatchIEnumerable(AzureTableDefaultPolicies.MAX_BULK_UPDATE_ROWS))
+                {
+                    tasks.Add(storage.DeleteTableEntriesAsync(batch));
+                }
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+            }
+            return entriesList.Count;
+        }
+
+
+        /// <summary>
+        /// Try once to conditionally update a data entry in the Azure table. Returns null if etag does not match.
+        /// </summary>
+        private async Task<string> TryUpdateTableEntryAsync(GossipTableEntry data, string dataEtag, [CallerMemberName]string operation = null)
```

Comment:
In two places I found where `TryUpdateTableEntryAsync` is called the returned value is only checked for null:

```csharp
 return (await TryUpdateTableEntryAsync(entry, eTag).ConfigureAwait(false) != null);
```
Since the string value is never used, why not return a bool instead?